### PR TITLE
Combined transaction base characteristics and Delete relevant SecretStore files thoroughly

### PR DESCRIPTION
**Pull request by chaychen2005**
Core changes:
1、Combined transaction base characteristics
2、Delete relevant SecretStore files thoroughly
3、Delete EIP155 logic

### CODE REVIEW

File: libethcore/TransactionBase.h [hunk 1]
```diff
@@ -237,20 +256,21 @@ class TransactionBase
...
-                   ///< creation transaction.
+    u256 m_blockLimit;  ///< The latest block number to be packaged for transaction.
+    bytes m_data;       ///< The data associated with the transaction, or the initialiser if it's a
+                        ///< creation transaction.
     boost::optional<SignatureStruct> m_vrs;  ///< The signature of the transaction. Encodes the
                                              ///< sender.
     int m_chainId = -4;                      ///< EIP155 value for calculating transaction hash
                                              ///< https://github.com/ethereum/EIPs/issues/155
-
+    u256 m_importtime = 0;     ///< The utc time at which a transaction enters the queue.
```
Comment (wheatli) [inline libethcore/TransactionBase.h]: Pls abey the code style

File: libethcore/Common.h [hunk 2]
```diff
@@ -218,9 +218,10 @@ struct TransactionSkeleton
     Address to;
     u256 value;
     bytes data;
-    u256 nonce = Invalid256;
+    u256 randomid = Invalid256;
```
[the remark this answers is not in the document]
Comment (chaychen2005) [reply libethcore/Common.h]: Done. Use nonce again and add its instructions.

File: libethcore/TransactionBase.cpp [hunk 3]
```diff
@@ -53,22 +54,23 @@ TransactionBase::TransactionBase(bytesConstRef _rlpData, CheckTransaction _check
...
                 InvalidTransactionFormat() << errinfo_comment("transaction RLP must be a list"));
 
-        m_nonce = rlp[0].toInt<u256>();
+        m_randomid = rlp[0].toInt<u256>();
         m_gasPrice = rlp[1].toInt<u256>();
         m_gas = rlp[2].toInt<u256>();
-        m_type = rlp[3].isEmpty() ? ContractCreation : MessageCall;
-        m_receiveAddress = rlp[3].isEmpty() ? Address() : rlp[3].toHash<Address>(RLP::VeryStrict);
-        m_value = rlp[4].toInt<u256>();
+        m_blockLimit = rlp[3].toInt<u256>();  // 3
```
[the remark this answers is not in the document]
Comment (chaychen2005) [reply libethcore/TransactionBase.cpp]: Done. Place blockLimit as last item already.

File: libethcore/TransactionBase.cpp [hunk 4]
```diff
@@ -158,8 +160,8 @@ void TransactionBase::streamRLP(RLPStream& _s, IncludeSignature _sig, bool _forE
     if (m_type == NullTransaction)
         return;
 
-    _s.appendList((_sig || _forEip155hash ? 3 : 0) + 6);
-    _s << m_nonce << m_gasPrice << m_gas;
+    _s.appendList((_sig || _forEip155hash ? 3 : 0) + 7);
```
[the remark this answers is not in the document]
Comment (chaychen2005) [reply libethcore/TransactionBase.cpp]: Three steps.First, the FISCO`s logic is merged, then the redundant logic is removed, and then the extension is considered again. Pls wait for the next submission.

File: libethcore/TransactionBase.h (hunk 1 above)
Comment (chaychen2005) [reply libethcore/TransactionBase.h]: Done，change it to m_importTime.

File: libethcore/TransactionBase.cpp [hunk 5]
```diff
@@ -158,7 +161,7 @@ void TransactionBase::streamRLP(RLPStream& _s, IncludeSignature _sig, bool _forE
     if (m_type == NullTransaction)
         return;
 
-    _s.appendList((_sig || _forEip155hash ? 3 : 0) + 6);
+    _s.appendList((_sig || _forEip155hash ? 3 : 0) + 7);
```
Comment (wheatli) [inline libethcore/TransactionBase.cpp]: drop _forEip155hash logic

File: libethcore/TransactionBase.h [hunk 6]
```diff
@@ -237,20 +256,24 @@ class TransactionBase
...
+                                    ///< transaction.
     u256 m_value;  ///< The amount of ETH to be transferred by this transaction. Called 'endowment'
                    ///< for contract-creation transactions.
     Address m_receiveAddress;  ///< The receiving address of the transaction.
     u256 m_gasPrice;           ///< The base fee and thus the implied exchange rate of ETH to GAS.
     u256 m_gas;  ///< The total gas to convert, paid for from sender's account. Any unused gas gets
                  ///< refunded once the contract is ended.
-    bytes m_data;  ///< The data associated with the transaction, or the initialiser if it's a
-                   ///< creation transaction.
+    u256 m_blockLimit;  ///< The latest block number to be packaged for transaction.
```
Comment (wheatli) [inline libethcore/TransactionBase.h]: place this new member near the m_importTime

File: libethcore/TransactionBase.cpp (hunk 5 above)
Comment (chaychen2005) [reply libethcore/TransactionBase.cpp]: Delete all EIP155`s logic

File: libethcore/TransactionBase.h (hunk 6 above)
Comment (chaychen2005) [reply libethcore/TransactionBase.h]: Done

File: libethcore/SealEngine.cpp [hunk 7]
```diff
@@ -93,15 +93,11 @@ void SealEngineFace::verifyTransaction(ImportRequirements::value _ir, Transactio
     BlockHeader const& _header, u256 const& _gasUsed) const
 {
     if ((_ir & ImportRequirements::TransactionSignatures) &&
-        _header.number() < chainParams().EIP158ForkBlock && _t.isReplayProtected())
-        BOOST_THROW_EXCEPTION(InvalidSignature());
-
-    if ((_ir & ImportRequirements::TransactionSignatures) &&
-        _header.number() < chainParams().experimentalForkBlock && _t.hasZeroSignature())
+        _header.number() < chainParams().experimentalForkBlock)
```
Comment (wheatli) [inline libethcore/SealEngine.cpp]: Pls check this logic again, it's not right

File: libethcore/SealEngine.cpp [hunk 8]
```diff
@@ -93,15 +93,11 @@ void SealEngineFace::verifyTransaction(ImportRequirements::value _ir, Transactio
...
         BOOST_THROW_EXCEPTION(InvalidSignature());
 
     if ((_ir & ImportRequirements::TransactionBasic) &&
-        _header.number() >= chainParams().experimentalForkBlock && _t.hasZeroSignature() &&
+        _header.number() >= chainParams().experimentalForkBlock &&
```
Comment (wheatli) [inline libethcore/SealEngine.cpp]: Pls think about the transaction checking logic in rounded system,  I think there are some useless checking

File: libethcore/TransactionBase.cpp [hunk 9]
```diff
@@ -73,25 +73,13 @@ TransactionBase::TransactionBase(bytesConstRef _rlpData, CheckTransaction _check
...
-                m_chainId = -4;
-            else
-                BOOST_THROW_EXCEPTION(InvalidSignature());
-
-            m_vrs = SignatureStruct{r, s, static_cast<byte>(v - (m_chainId * 2 + 35))};
-
-            if (_checkSig >= CheckTransaction::Cheap && !m_vrs->isValid())
-                BOOST_THROW_EXCEPTION(InvalidSignature());
-        }
+        if ((v != 27) && (v != 28))
```
Comment (wheatli) [inline libethcore/TransactionBase.cpp]: Pls add some comments about these magic number, what do they means?

File: libethcore/TransactionBase.cpp [hunk 10]
```diff
@@ -156,12 +139,12 @@ void TransactionBase::sign(Secret const& _priv)
...
 }
 
-void TransactionBase::streamRLP(RLPStream& _s, IncludeSignature _sig, bool _forEip155hash) const
+void TransactionBase::streamRLP(RLPStream& _s, IncludeSignature _sig) const
 {
     if (m_type == NullTransaction)
         return;
 
-    _s.appendList((_sig || _forEip155hash ? 3 : 0) + 7);
+    _s.appendList((_sig ? 3 : 0) + 7);
```
Comment (wheatli) [inline libethcore/TransactionBase.cpp]: Also we need some comments here, what's the meaning of 3 and 7 ?

File: libethcore/SealEngine.cpp (hunk 7 above)
Comment (chaychen2005) [reply libethcore/SealEngine.cpp]: Verify with Tab that this part of the logic involving forks is removable.

File: libethcore/SealEngine.cpp (hunk 8 above)
Comment (chaychen2005) [reply libethcore/SealEngine.cpp]: Verify with Tab that this part of the logic involving forks is removable.

File: libethcore/TransactionBase.cpp [hunk 11]
```diff
@@ -188,15 +188,6 @@ int64_t TransactionBase::baseGasRequired(
     return g;
 }
 
-bigint TransactionBase::gasRequired(
```
Comment (wheatli) [inline libethcore/TransactionBase.cpp]: Can we drop this logic? I have a remember that it will be used somewhere, pls check it

Comment (wheatli) [reply libethcore/TransactionBase.cpp]: Okay, my fault, we can use basicGasRequired

**Pull request merged**